### PR TITLE
Eng 1137/38 candy data editor candy float/candy principal components

### PR DESCRIPTION
**Pull request by aletex1994**
**FLOAT FORM**
- Added component floatForm.tsx
- Added FloatForm > converter.ts 
- Updated constants.ts
- wrote unit tests for FloatForm > converter.ts > convertToCandyFloat()
- updated formTypes > index.tsx

**PRINCIPAL FORM**
- Added component principalForm.tsx
- Added principalForm > converter.ts 
- Updated constants.ts
- wrote unit tests for PrincipalForm > converter.ts > convertToPrincipal()
- updated formTypes > index.tsx

### CODE REVIEW

File: packages/candy_editor/src/assets/formTypes/PrincipalForm/converters.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { CandyPrincipal } from '../../../types';
+import { Principal } from '@dfinity/principal';
+
+export function convertToCandyPrincipal(typedValue: string): CandyPrincipal | undefined {
+    try {
+        const principal = Principal.fromText(typedValue);
+        return { Principal: principal };
+    } catch (e) {
```
Comment (Motokoder) [inline packages/candy_editor/src/assets/formTypes/PrincipalForm/converters.ts:8]: Please run linter and fix any errors.

File: packages/candy_editor/src/assets/formTypes/FloatForm/converters.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { CandyFloat } from '../../../types';
+
+export function convertToCandyFloat(typedValue: string): CandyFloat | undefined {
+    if (/^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/.test(typedValue)) {
```
Comment (Motokoder) [inline packages/candy_editor/src/assets/formTypes/FloatForm/converters.ts:4]: Needs a range check.

**Review by Motokoder — CHANGES_REQUESTED** [review]

File: packages/candy_editor/src/assets/formTypes/FloatForm/converters.ts (hunk 2 above)
Comment (aletex1994) [reply packages/candy_editor/src/assets/formTypes/FloatForm/converters.ts]: https://origyn.atlassian.net/jira/software/projects/ENG/boards/37/roadmap?selectedIssue=ENG-1264

Comment (Motokoder) [reply packages/candy_editor/src/assets/formTypes/FloatForm/converters.ts]: Alessandro to create a bug and work on this later.

**Review by Motokoder — APPROVED** [review]

**Pull request merged**